pkg/cloud/openstack/services/loadbalancer/loadbalancer.go

```diff
@@ -264,6 +264,39 @@ func (s *Service) ReconcileLoadBalancerMember(clusterName string, machine *clust
 	return nil
 }
 
+func (s *Service) DeleteLoadBalancer(clusterName string, clusterProviderSpec *providerv1.OpenstackClusterProviderSpec, clusterProviderStatus *providerv1.OpenstackClusterProviderStatus) error {
+	loadBalancerName := fmt.Sprintf("%s-cluster-%s-%s", networkPrefix, clusterName, kubeapiLBSuffix)
+	lb, err := checkIfLbExists(s.loadbalancerClient, loadBalancerName)
+	if err != nil {
+		return err
+	}
+	deleteOpts := loadbalancers.DeleteOpts{
+		Cascade: true,
+	}
+	klog.Infof("Deleting loadbalancer %s", loadBalancerName)
+	err = loadbalancers.Delete(s.loadbalancerClient, lb.ID, deleteOpts).ExtractErr()
+	if err != nil {
+		return fmt.Errorf("error deleting loadbalancer: %s", err)
+	}
+
+	// floating ip
+	if clusterProviderSpec.APIServerLoadBalancerFloatingIP != "" {
```

Comment:
What should we do in cases a already existing floating ip is just associated? In my case that's not a problem because I can delete the floating ip. But what if that's forbidden?

Reply:
that's a good question...can we distinguish those 2 cases? I don't know how to check it because it's already delete stage, should we create another flag?

Reply:
I think we should avoid adding even more flags, because it get's problematic from a UX perspective.

I have 2 ideas:
1. [Not sure if that's possible] Ask Keystone if we're allowed to delete the floating ip and only do it if it's allowed
2. Try to delete the floating ip and ignore the "you're not allowed to" (not sure how it's phrased" error

In both variants we should probably for disassociate the floating ip (which is an update on the floating ip with an empty port_id, similar to what I did here: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/435/commits/79cd5abd2f1fff965c5ea08f1e42eb71c082f933#diff-1144c99c6890d759d38d9549258a187aR84)